container plugin: segmentation violation when running on Falco with `-o base_syscalls.all=true`

**Describe the bug**

See the screenshot. 

**How to reproduce it**

Try 
```
sudo falco -o libs_logger.severity=trace -o libs_logger.enabled=true -o base_syscalls.all=true
```

Try it multiple times if the problem does not occur. There's likely a race condition under the hood.

**Expected behaviour**

<!-- A clear and concise description of what you expected to happen. -->

**Screenshots**

```
Tue Aug 26 18:14:04 2025: [libs]: container: starting async go-worker
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x120 pc=0x7fc480ce0dfb]

goroutine 17 [running, locked to thread]:
github.com/falcosecurity/plugins/plugins/container/go-worker/pkg/container.(*dockerEngine).ctrToInfo(_, {_, _}, {0x0, {0x0, 0x0, 0x0}, 0x0, 0x0, 0x0})
        /home/leogr/code/github.com/falcosecurity/plugins/plugins/container/go-worker/pkg/container/docker.go:107 +0x5b
github.com/falcosecurity/plugins/plugins/container/go-worker/pkg/container.(*dockerEngine).List(0xc0002a0330, {0x7fc4815e5440, 0xc0004b2000})
        /home/leogr/code/github.com/falcosecurity/plugins/plugins/container/go-worker/pkg/container/docker.go:357 +0x41a
main.StartWorker(0x7fc47febe700, 0x7fc47f7fd800, 0x7ffe9ea582b8)
        /home/leogr/code/github.com/falcosecurity/plugins/plugins/container/go-worker/worker_api.go:77 +0x4f3
[1]    350927 IOT instruction  sudo falco -o libs_logger.severity=trace -o libs_logger.enabled=true -o
```

**Environment**

<!-- Please complete the following info. -->

- Falco version:
```
Tue Aug 26 18:17:24 2025: Falco version: 0.41.3 (x86_64)
Tue Aug 26 18:17:24 2025: Falco initialized with configuration files:
Tue Aug 26 18:17:24 2025:    /etc/falco/config.d/falco.container_plugin.yaml | schema validation: ok
Tue Aug 26 18:17:24 2025:    /etc/falco/falco.yaml | schema validation: ok
Tue Aug 26 18:17:24 2025: System info: Linux version 6.16.3-arch1-1 (linux@archlinux) (gcc (GCC) 15.2.1 20250813, GNU ld (GNU Binutils) 2.45.0) #1 SMP PREEMPT_DYNAMIC Sat, 23 Aug 2025 15:32:49 +0000
Falco version: 0.41.3
Libs version:  0.21.0
Plugin API:    3.11.0
Engine:        0.50.0
Driver:
  API version:    8.0.0
  Schema version: 3.6.0
  Default driver: 8.1.0+driver
```
- System info:
<!-- Falco has a built-in support command you can use  "falco --support | jq .system_info" -->
- Cloud provider or hardware configuration:
- OS:
<!-- Eg., output of "cat /etc/os-release". -->
- Kernel:
<!-- Eg., output of "uname -a". -->
- Installation method:
<!-- Eg., Kubernetes, RPM, DEB, from source? -->

**Additional context**

I reproduced it both with plugin 0.3.3 and the latest dev version.

--- a/plugins/container/CMakeLists.txt
+++ b/plugins/container/CMakeLists.txt
@@ -9,7 +9,7 @@ list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/cmake/modules")
 # project metadata
 project(
         container
-        VERSION 0.3.6
+        VERSION 0.3.7
         DESCRIPTION "Falco container metadata enrichment Plugin"
         LANGUAGES CXX)
 

--- a/plugins/container/go-worker/pkg/container/cri.go
+++ b/plugins/container/go-worker/pkg/container/cri.go
@@ -344,7 +344,7 @@ func (c *criEngine) ctrToInfo(ctx context.Context, ctr *v1.ContainerStatus, podS
 		imageID = ctr.GetImageId()
 	}
 
-	return event.Info{
+	evtInfo := event.Info{
 		Container: event.Container{
 			Type:             c.runtime,
 			ID:               shortContainerID(ctr.Id),
@@ -363,10 +363,6 @@ func (c *criEngine) ctrToInfo(ctx context.Context, ctr *v1.ContainerStatus, podS
 			CreatedTime:      nanoSecondsToUnix(ctr.CreatedAt),
 			Env:              ctrInfo.getEnvs(),
 			FullID:           ctr.Id,
-			HostIPC:          podSandboxStatus.Linux.Namespaces.Options.Ipc == v1.NamespaceMode_NODE,
-			HostNetwork:      podSandboxStatus.Linux.Namespaces.Options.Network == v1.NamespaceMode_NODE,
-			HostPID:          podSandboxStatus.Linux.Namespaces.Options.Pid == v1.NamespaceMode_NODE,
-			Ip:               podSandboxStatus.Network.Ip,
 			IsPodSandbox:     isPodSandbox,
 			Labels:           labels,
 			MemoryLimit:      memoryLimit,
@@ -378,6 +374,19 @@ func (c *criEngine) ctrToInfo(ctx context.Context, ctr *v1.ContainerStatus, podS
 			Size:             size,
 		},
 	}
+
+	if podSandboxStatus.Linux != nil && podSandboxStatus.Linux.Namespaces != nil &&
+		podSandboxStatus.Linux.Namespaces.Options != nil {
+		evtInfo.HostIPC = podSandboxStatus.Linux.Namespaces.Options.Ipc == v1.NamespaceMode_NODE
+		evtInfo.HostNetwork = podSandboxStatus.Linux.Namespaces.Options.Network == v1.NamespaceMode_NODE
+		evtInfo.HostPID = podSandboxStatus.Linux.Namespaces.Options.Pid == v1.NamespaceMode_NODE
+	}
+
+	if podSandboxStatus.Network != nil {
+		evtInfo.Ip = podSandboxStatus.Network.Ip
+	}
+
+	return evtInfo
 }
 
 func (c *criEngine) get(ctx context.Context, containerId string) (*event.Event, error) {

--- a/plugins/container/go-worker/pkg/container/docker.go
+++ b/plugins/container/go-worker/pkg/container/docker.go
@@ -5,6 +5,10 @@ import (
 	"context"
 	"encoding/json"
 	"errors"
+	"strings"
+	"sync"
+	"time"
+
 	"github.com/docker/docker/api/types/container"
 	"github.com/docker/docker/api/types/events"
 	"github.com/docker/docker/api/types/filters"
@@ -12,9 +16,6 @@ import (
 	"github.com/docker/docker/client"
 	"github.com/falcosecurity/plugins/plugins/container/go-worker/pkg/config"
 	"github.com/falcosecurity/plugins/plugins/container/go-worker/pkg/event"
-	"strings"
-	"sync"
-	"time"
 )
 
 const k8sLastAppliedConfigLabel = "io.kubernetes.container.last-applied-config"
@@ -352,10 +353,11 @@ func (dc *dockerEngine) List(ctx context.Context) ([]event.Event, error) {
 				},
 				IsCreate: true,
 			}
-		}
-		evts[idx] = event.Event{
-			Info:     dc.ctrToInfo(ctx, ctrJson),
-			IsCreate: true,
+		} else {
+			evts[idx] = event.Event{
+				Info:     dc.ctrToInfo(ctx, ctrJson),
+				IsCreate: true,
+			}
 		}
 	}
 	return evts, nil

